tools/src/main/scala/org/apache/spark/sql/rapids/tool/EventProcessorBase.scala

```diff
@@ -0,0 +1,211 @@
+      case _: SparkListenerApplicationEnd =>
+        doSparkListenerApplicationEnd(app,
+          event.asInstanceOf[SparkListenerApplicationEnd])
+      case _: SparkListenerExecutorAdded =>
+        doSparkListenerExecutorAdded(app,
+          event.asInstanceOf[SparkListenerExecutorAdded])
+      case _: SparkListenerExecutorRemoved =>
+        doSparkListenerExecutorRemoved(app,
+          event.asInstanceOf[SparkListenerExecutorRemoved])
+      case _: SparkListenerTaskStart =>
+        doSparkListenerTaskStart(app,
+          event.asInstanceOf[SparkListenerTaskStart])
+      case _: SparkListenerTaskEnd =>
+        doSparkListenerTaskEnd(app,
+          event.asInstanceOf[SparkListenerTaskEnd])
+      case _: SparkListenerSQLExecutionStart =>
+        doSparkListenerSQLExecutionStart(app,
+          event.asInstanceOf[SparkListenerSQLExecutionStart])
+      case _: SparkListenerSQLExecutionEnd =>
+        doSparkListenerSQLExecutionEnd(app,
+          event.asInstanceOf[SparkListenerSQLExecutionEnd])
+      case _: SparkListenerDriverAccumUpdates =>
+        doSparkListenerDriverAccumUpdates(app,
+          event.asInstanceOf[SparkListenerDriverAccumUpdates])
+      case _: SparkListenerJobStart =>
+        doSparkListenerJobStart(app,
+          event.asInstanceOf[SparkListenerJobStart])
+      case _: SparkListenerJobEnd =>
+        doSparkListenerJobEnd(app,
+          event.asInstanceOf[SparkListenerJobEnd])
+      case _: SparkListenerStageSubmitted =>
+        doSparkListenerStageSubmitted(app,
+          event.asInstanceOf[SparkListenerStageSubmitted])
+      case _: SparkListenerStageCompleted =>
+        doSparkListenerStageCompleted(app,
+          event.asInstanceOf[SparkListenerStageCompleted])
+      case _: SparkListenerTaskGettingResult =>
+        doSparkListenerTaskGettingResult(app,
+          event.asInstanceOf[SparkListenerTaskGettingResult])
+      case _: SparkListenerSQLAdaptiveExecutionUpdate =>
+        doSparkListenerSQLAdaptiveExecutionUpdate(app,
+          event.asInstanceOf[SparkListenerSQLAdaptiveExecutionUpdate])
+      case _: SparkListenerSQLAdaptiveSQLMetricUpdates =>
+        doSparkListenerSQLAdaptiveSQLMetricUpdates(app,
+          event.asInstanceOf[SparkListenerSQLAdaptiveSQLMetricUpdates])
+      case _ =>
+        val wasResourceProfileAddedEvent = doSparkListenerResourceProfileAddedReflect(app, event)
+        if (!wasResourceProfileAddedEvent) doOtherEvent(app, event)
+    }
+  }
+
+  def doSparkListenerResourceProfileAddedReflect(
+      app: T,
+      event: SparkListenerEvent): Boolean = {
+    val rpAddedClass = "org.apache.spark.scheduler.SparkListenerResourceProfileAdded"
+    if (event.getClass.getName.equals(rpAddedClass)) {
+      try {
+        event match {
+          case _: SparkListenerResourceProfileAdded =>
+            doSparkListenerResourceProfileAdded(app,
+              event.asInstanceOf[SparkListenerResourceProfileAdded])
```

Comment:
nit: can avoid asInstanceOf by using a variable in the pattern instead of `_`

Reply:
updated